app/decorators/current_user.rb

```diff
@@ -0,0 +1,46 @@
+class CurrentUser < Draper::Decorator
+  decorates :user
+  delegate_all
+
+  def session_url
+    if object
+      h.link_to("Logout",
+                h.logout_path,
+                method: :delete)
+    else
+      h.link_to("Login with GitHub",
+                "/auth/github",
+                method: :get)
+    end
+  end
+
+  def create_room_url
+    if object && !h.current_page?(h.new_room_path)
+      h.link_to("Create New Room",
+                h.new_room_path)
+    else
+      ""
+    end
+  end
+
+  def form_link(form)
+    if object
+      form.submit("Create room", class: "btn")
+    else
+      h.link_to("Login to create room",
+                "/auth/github",
+                class: "btn")
+    end
+  end
+
+  def close_topic(room)
+    if object == room.user
+      h.button_to("Close this Topic",
+                 h.room_path(room),
+                 method: :put,
+                 class: "btn col s12")
+    else
+      ""
+    end
+  end
+end
```

Comment:
I do feel like it's a better approach. In the view there is one line with only one method which is easy to navigate to. Yeah else can be nil too, don't know why I thought ```""``` was necessary. h is short for Rails View Helpers such as link_to, form_for, etc. I don't like the voodoo Draper does to gain access to those in the decorator, but I think Rails is already doing enough magic for this to be justifiable.